src/operations/monochrome.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{error::MagickError, image::Image, wm_err};
+use image::{DynamicImage, GrayImage, Luma};
+
+pub fn monochrome(image: &mut Image) -> Result<(), MagickError> {
+    let mut grayscale = image.pixels.to_luma8();
+    apply_contrast(&mut grayscale, CONTRAST_FACTOR);
+    let noise_texture = NoiseTexture::load()?;
+    apply_dithering(&mut grayscale, &noise_texture);
+    image.pixels = DynamicImage::ImageLuma8(grayscale);
+
+    Ok(())
+}
+
+/// This is empirically tuned to give results similar to ImageMagick's -monochrome
+const CONTRAST_FACTOR: f32 = 10.0;
+
+fn apply_contrast(image: &mut GrayImage, contrast_factor: f32) {
+    let offset = 128.0 * (1.0 - contrast_factor);
+
+    for pixel in image.pixels_mut() {
+        for channel in pixel.0.iter_mut() {
+            let value = *channel as f32;
+            let adjusted = (value * contrast_factor + offset).clamp(0.0, 255.0);
+            *channel = adjusted as u8;
+        }
+    }
+}
+
+const BACKGROUND: Luma<u8> = Luma([255]);
+const FOREGROUND: Luma<u8> = Luma([0]);
+const DITHER_THRESHOLD: u8 = 60;
+
+fn apply_dithering(image: &mut GrayImage, noise_texture: &NoiseTexture) {
+    let width = image.width();
+    let height = image.height();
+
+    for y in 0..height {
+        for x in 0..width {
+            let pixel_luma = image.get_pixel(x, y).0[0];
+
+            if pixel_luma >= (255 - DITHER_THRESHOLD) || pixel_luma <= (0 + DITHER_THRESHOLD) {
+                continue;
+            }
+
+            let noise_luma = noise_texture.get(x, y);
+            let color = if pixel_luma > noise_luma {
+                BACKGROUND
+            } else {
+                FOREGROUND
+            };
+
+            image.put_pixel(x, y, color);
+        }
+    }
+}
+
+/// Blue noise texture data
+pub struct NoiseTexture {
+    data: Vec<u8>,
+    width: usize,
+    height: usize,
+}
+
+impl NoiseTexture {
+    /// Load blue noise texture from a file
+    pub fn load() -> Result<Self, MagickError> {
+        let path_to_crate = env!("CARGO_MANIFEST_DIR");
+
+        let img = image::open(format!("{}/src/operations/blue-noise.png", path_to_crate))
+            .map_err(|e| wm_err!("failed to load blue noise texture: {}", e))?;
```

Comment:
Loading at runtime is a no-go, we need a self-contained binary. Also, PNG feature may be disabled when building `wondermagick`, which would cause this image to fail to load.

Instead, please use `include_bytes!` on an uncompressed luma8 data instead of a PNG image. 64x64 pixels should be enough, and we can easily afford that - that's just 4Kb of data.

This is how you turn a `&'static [u8]` into a `GrayImage`: https://docs.rs/image/latest/image/type.GrayImage.html#method.from_raw